build-tools/packages/build-cli/src/library/repoPolicyCheck/npmPackages.ts

```diff
@@ -1965,6 +1966,69 @@ export const handlers: Handler[] = [
 				}
 			});
 
+			return result;
+		},
+	},
+	{
+		name: "npm-check-unused-dependencies",
+		match,
+		handler: async (file: string): Promise<string | undefined> => {
+			const depcheckConfigFileName = ".depcheckrc.cjs";
+			const packageDir = path.resolve(path.dirname(file));
+			const depcheckConfigFilePath = path.resolve(
+				path.join(packageDir, depcheckConfigFileName),
+			);
+			const configExists = fs.existsSync(depcheckConfigFilePath);
+			let options: depcheck.Options = {};
+			if (configExists) {
+				try {
+					options = require(depcheckConfigFilePath) as depcheck.Options;
+				} catch (error) {
+					console.log(`Error reading ${depcheckConfigFileName} file for ${packageDir}`, error);
+					return;
+				}
+			}
+			try {
+				const result = await depcheck(packageDir, options);
+				const packageErrors: string[] = [];
+				if (result.devDependencies.length > 0) {
+					packageErrors.push(`[Unused devDependencies]:${result.devDependencies.join(",")}`);
+				}
+				if (result.dependencies.length > 0) {
+					packageErrors.push(`[Unused dependencies]:${result.dependencies.join(",")}`);
+				}
+				return packageErrors.length > 0 ? packageErrors.join(newline) : undefined;
+			} catch (error) {
+				return `Error running depcheck for ${packageDir}: ${error}`;
+			}
+		},
+		resolver: async (
+			file: string,
+			_root: string,
+			handlerOutput: string,
```

Comment:
Could you void the extra processing of the string handler output by instead passing an object? For example, you could change the handler to return a the string error array AND an `unknown` object. Then pass the unknown object to the resolver, which can cast it to what it expects reasonably safely. That should allow your resolver to re-use the data from the depcheck response without parsing strings in the resolver.